db_master_deploy/deploy.sh

```diff
@@ -370,6 +371,52 @@ copy_table() {
     fi
 }
 
+#######################################
+# write lock
+# Globals:
+#   target_db
+#   COMMAND
+#   USER
+# Arguments:
+#   None
+# Returns:
+#   None
+#######################################
+write_lock() {
+    local lockfile="${LOCK_DIR}/${target_db}.lock"
+    # if target db is locked enter loop and wait for lockfile
+    until [ ! -f ${lockfile} ]
+    do
+        echo "target db ${target_db} is locked, waiting for deploy process to finish"
+        sleep 5
+    done
```

Comment:
the max retry value depends on the maximum time needed for a db or table deplo (db duplication, table copy on the master instance and waiting for the slaves to be in sync). 
this value will change in future and it is not stable. i would prefer to not hardcode a max retry value in this script.